astilectron.go

```diff
@@ -182,6 +185,48 @@ func (a *Astilectron) provision() error {
 	return a.provisioner.Provision(ctx, a.options.AppName, runtime.GOOS, runtime.GOARCH, *a.paths)
 }
 
+// function to create TCP/Unix socket connection
+func (a *Astilectron) establishConnection() (err error) {
+	/*
+	 * Switching between Unix-Socket and TCP-Socket
+	 * Windows will use TCP Socket
+	 * MAC and Linux will use Unix-Socket
+	 */
+	if runtime.GOOS == "windows" {
+		// Unfortunately communicating with Electron through stdin/stdout doesn't work on Windows so all communications
+		// will be done through TCP
+		if err = a.listenTCP(); err != nil {
+			return errors.Wrap(err, "TCP Socket listening failed")
+		}
+	} else {
+		if err = a.listenUnixSoc(); err != nil {
+			return errors.Wrap(err, "Unix Socket listening failed")
+		}
+	}
+	return nil
+}
+
+// Creates a unix socket
+func (a *Astilectron) listenUnixSoc() (err error) {
+	// Log
+	astilog.Debug("Listening...")
+
+	_ = os.Remove(UNIX_SOCKET_PATH)
```

Comment:
Could you check the error? If you're worried about getting a "file doesn't exist" error, the condition should be `if err != nil && !os.IsNotExist(err)`